From 685c2b0adccea82137bd114d4151dc490d187e21 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 24 Feb 2022 09:49:16 +0200
Subject: [PATCH 1/5] s32: Introduce FIP_HDR_SIZE makefile variable

FIP_HDR_SIZE is defined to allow standalone build of the BL2 stage.
This is needed because BL2 stage needs to know the size of the FIP
header to determine the address of the device tree and FIP header.

This allows to compile BL2 for RDB2 using the command:
make PLAT=s32g274ardb2 FIP_SD_OFFSET=0x1000 FIP_HDR_SIZE=0x200 bl2

Issue: ALB-8551
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_common.mk | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index 9eb2aaaf2..824a6390c 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -262,10 +262,6 @@ ${DUMMY_FIP}: fiptool ${DUMMY_STAGE} | ${BUILD_PLAT}
 	${Q}$(call update_fip, ${DUMMY_STAGE}, ${DUMMY_STAGE}, "$@_temp")
 	${Q}mv "$@_temp" $@
 
-${FIP_HDR_SIZE_FILE}: ${DUMMY_FIP} ${FIPTOOL}
-	${Q}${ECHO} "  CREATE  $@"
-	${Q}$(call get_fip_hdr_size, ${DUMMY_FIP}) > $@
-
 ${DUMMY_FIP_S32}: ${DUMMY_FIP}
 	${Q}${ECHO} "  MKIMAGE $@"
 	${Q}$(call run_mkimage, ${BL2_BASE}, ${BL2_BASE}, ${MKIMAGE_CFG}, $<, $@) 2> /dev/null
@@ -286,6 +282,17 @@ ${FIP_OFFSET_FILE}: ${DUMMY_FIP_S32}
 	${Q}OFF=$$(${MKIMAGE} -l $< 2>&1 | grep Application | awk '{print $$3}');\
 	$(call save_fip_off, $${OFF},$@)
 
+ifndef FIP_HDR_SIZE
+${FIP_HDR_SIZE_FILE}: ${DUMMY_FIP} ${FIPTOOL}
+	${Q}${ECHO} "  CREATE  $@"
+	${Q}$(call get_fip_hdr_size, ${DUMMY_FIP}) > $@
+else
+${FIP_HDR_SIZE_FILE}: FORCE
+	${Q}${ECHO} "  CREATE  $@"
+	${Q}${ECHO} "${FIP_HDR_SIZE}" > "$@"
+
+endif
+
 ifeq ($(FIP_SD_OFFSET)$(FIP_QSPI_OFFSET)$(FIP_MEMORY_OFFSET)$(FIP_EMMC_OFFSET),)
 ${FIP_SD_OFFSET_FILE}: ${IVT_LOCATION_FILE} ${FIP_OFFSET_FILE}
 	${Q}${ECHO} "  CREATE  $@"
-- 
2.17.1

